### ISSUE TITLE
The libtary is linked with likokkos even when configured with DEAL_II_WITH_KOKKOS=OFF

### ISSUE THREAD

**Issue by yurivict** [dealii/dealii#15690]
Version: 9.5.0
FreeBSD 13.2

**Comment by RichardYCJ** [comment]
According to the [release page](https://www.dealii.org/current/doxygen/deal.II/changes_between_9_4_0_and_9_5_0.html#:~:text=New%3A%20Kokkos%20is%20now%20a%20required%20dependency), `kokkos` is a required dependency after version 9.5.0

**Comment by drwells** [comment]
Yes, Kokkos is required as of 9.5.0 (and we have a bundled version in case people do not have it installed).

It sounds like the error message printed when someone provides `-DDEAL_II_WITH_KOKKOS=OFF` is wrong - we should fix that.

**Comment by yurivict** [comment]
Does ```DEAL_II_WITH_KOKKOS=OFF``` disable anything at all?

**Comment by tamiko** [comment]
@drwells @yurivict Correct, there is simply no error message.
Neither for `DEAL_II_WITH_KOKKOS`, nor for `DEAL_II_WITH_BOOST`.
This is due to
https://github.com/dealii/dealii/blob/01b805b00b621143e6ca89ff13dfde00807e0e31/cmake/configure/configure_20_boost.cmake#L30-L32
which implies that we never trigger:
https://github.com/dealii/dealii/blob/01b805b00b621143e6ca89ff13dfde00807e0e31/cmake/configure/configure_20_boost.cmake#L179-L191

So this is "wrong" for about 11 years. Same for Kokkos as the logic was simply copied.
We could simply remove the second code block and issue a warning when we switch the flag to ON.
